syntax = "proto2";

// DataFed version numbers must be manually updated as needed.
// All minor version numbers should be reset when VER_MAJOR is incremented.
// VER_MAPI_MINOR should be reset to 0 when VER_MAPI_MAJOR is incremented.
// Servers and clients must be interoperable as long as VER_MAJOR and VER_MAPI_MAJOR match

// Minor version are used to notify admins when local servers need to be updated.
// This is accomplished by sending a version request to the Core server on startup
// and comparing to the local version numbers.

enum Version
{
    option allow_alias = true; // Must be commented out if there are no duplicate values

    VER_MAJOR = 1;          // System MAJOR version, no backward compatibility
    VER_MAPI_MAJOR = 4;     // Message API MAJOR version, no backward compatibility
    VER_MAPI_MINOR = 1;     // Message API MINOR version, backward compatible
    VER_CORE = 0;           // Core server MINOR version, information only
    VER_WEB = 0;            // Web server MINOR version, info/notification purposes
    VER_REPO = 0;           // Repo server MINOR version, info/notification purposes
    VER_CLIENT_PY = 0;      // Python client/api MINOR version, info/notification purposes
}

